common/src/main/java/io/pravega/common/util/ResourcePool.java

```diff
@@ -281,6 +279,48 @@ public void shutdown() {
 
     @Data
     private static class WaitingRequest<T> {
-        private final CompletableFuture<T> future;
+        private final CompletableFuture<ClosableResource<T>> future;
+    }
+
+    /**
+     * A closeable resource wrapper class which returns the resource back to the pool automatically once it is closed.
+     * Its close method is idempotent and can be invoked multiple times without returning the resource more than once. 
+     * @param <T> Type of underlying resource
+     */
+    public static class ClosableResource<T> implements AutoCloseable {
```

Comment:
`Closable` -> `Closeable`. Even though my browser's spellcheck complains about this, I prefer to be consistent with `AutoCloseable`.

Reply:
plus you used `closeable` in line 286 :)